lib/oracle.js

```diff
@@ -31,22 +21,42 @@ exports.initialize = function initializeDataSource(dataSource, callback) {
     return;
   }
 
-  var s = dataSource.settings;
+  var s = dataSource.settings || {};
   var oracle_settings = {
-    hostname: s.host || s.hostname || 'localhost',
-    port: s.port || 1521,
+    connectString: s.connectString || s.url || s.tns,
     user: s.username || s.user,
     password: s.password,
-    database: s.database || 'XE',
     debug: s.debug || debug.enabled,
-    minConn: s.minConn || 1,
-    maxConn: s.maxConn || 10,
-    incrConn: s.incrConn || 1,
-    timeout: s.timeout || 10
+    poolMin: s.poolMin || s.minConn || 1,
+    poolMax: s.poolMax || s.maxConn || 10,
+    poolIncrement: s.poolIncrement || s.incrConn || 1,
+    poolTimeout: s.poolTimeout || s.timeout || 60,
+    isAutoCommit: s.isAutoCommit || s.autoCommit,
+    outFormat: oracle.OBJECT,
+    maxRows: s.maxRows || 100,
+    stmtCacheSize: s.stmtCacheSize || 30,
+    connectionClass: 'loopback-connector-oracle'
   };
-  if (s.tns) {
-    oracle_settings.tns = s.tns;
+
+  if (s.isAutoCommit === undefined) {
+    s.isAutoCommit = true; // Default to true
+  }
+
+  if (!s.connectString) {
+    var hostname = s.host || s.hostname || 'localhost';
+    var port = s.port || 1521;
+    var database = s.database || 'XE';
+    oracle_settings.connectString = '//' + hostname + ':' + port +
+      '/' + database;
```

Comment:
@raymondfeng I still think the `tns` and `url` wouldn't work here. 
I think we should check `oracle_settings.connectString` instead of `s.connectString` as following:
```js
var oracle_settings = {
    connectString: s.connectString || s.url || s.tns,
 };

// ... ...

if (!oracle_settings.connectString) {
  oracle_settings = generatedSettings;
}
```

Others LGTM